### PR TITLE
Build: add pre-commit for flake8

### PR DESCRIPTION
**Pull request by kinnou02**
I add to disable some warning, a few of them should be enable later but
they require a lot of change. Most of them are from star import...

A few unused import are infact used, I will resolve this in another PR

### CODE REVIEW

**Review by benoit-bst — APPROVED** [review]
:rainbow: Can avoid the "abort" :imp:

**Review by xlqian — COMMENTED** [review]
:+1:

**Review by xlqian — APPROVED** [review]

**Review by mbounabi — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ repos:
     -   id: trailing-whitespace
     -   id: check-ast
     -   id: check-merge-conflict
-    #-   id: flake8
+    -   id: flake8
```
Comment (pbougue) [inline .pre-commit-config.yaml:13]: Can you force python version (as it doesn't pass on python3.6)?
```suggestion
    -   id: flake8
        language_version: python2.7
```

File: source/jormungandr/requirements_dev.txt [hunk 2]
```diff
@@ -11,3 +11,4 @@ jsonschema==2.6.0
 pytest-timeout==1.3.3
 mypy==0.670; python_version > '3.0'
 mypy-extensions==0.4.1; python_version > '3.0'
+flake8==3.5.0
```
Comment (pbougue) [inline source/jormungandr/requirements_dev.txt:14]: Aren't they all supposed to be in a different `requirements_precommit.txt` file?

Comment (kinnou02) [reply source/jormungandr/requirements_dev.txt:14]: I don't see why, you should have them when you are developing as most editor will use mypy and flake to check the code while you write it.

**Review by pbougue — APPROVED** [review]
:+1: It's better than before

The other problems on precommit should be handled separately:
* as discussed: maybe separate precommit requirements (and include them into each requirement_dev)
* on short term: change scope of mypy to just source/jormungandr/jormungandr as travis only checks that (before including more and fixing `precommit run --all` )
* change scope of trailing-whitespace (exclude rst and md files)
* update travis to work using the precommit for real ;)

**Pull request merged**